### PR TITLE
log: More detailed doc around Wrapper interface

### PR DESCRIPTION
**Pull request by fishy**
Also add FallbackWrapper for nil-safe usage of it.

### CODE REVIEW

File: log/wrapper.go [hunk 1]
```diff
@@ -7,20 +7,53 @@ import (
...
+//
+// This interface is meant to solve Principle 2 above.
 // In reality we might actually use different logging libraries in different
 // services, and they are not always compatible with each other.
 // Wrapper is a simple common ground that it's easy to wrap whatever logging
 // library we use into.
 //
-// This is also the same type as thrift.Logger and can be used interchangeably
-// (sometimes with a typecast).
+// For production code,
```
Comment (pacejackson) [inline log/wrapper.go:29]: Should we also call out that this is really only meant for passing into libraries, production code can just use the functions provided by the `log` module?

Comment (fishy) [reply log/wrapper.go]: Done. PTAL?

**Comment by fishy** [comment]
:haircut:

I added even more clarification about when to use log.Wrapper.

I also did an audit on all the current use of it, and found out that we are fine :) But as a result I unexported Span.LogError to avoid it being abused by service code.

File: log/wrapper.go [hunk 2]
```diff
@@ -7,20 +7,72 @@ import (
...
 // services, and they are not always compatible with each other.
 // Wrapper is a simple common ground that it's easy to wrap whatever logging
 // library we use into.
 //
-// This is also the same type as thrift.Logger and can be used interchangeably
-// (sometimes with a typecast).
+// With that in mind, this interface should only be used by library code,
+// when the case satisfies all of the following 3 criteria:
+//
+// 1. This is a bad thing happened.
```
Comment (pacejackson) [inline log/wrapper.go:35]:
```suggestion
1. A bad thing happened.
```

File: log/wrapper.go [hunk 3]
```diff
@@ -7,20 +7,72 @@ import (
...
+//
+// 2. This is an unexpected thing happened.
```
Comment (pacejackson) [inline log/wrapper.go:37]:
```suggestion
// 2. This is unexpected.
```

**Pull request merged**